pkg/frontend/adminactions/delete_managedresource.go

```diff
@@ -0,0 +1,53 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/loadbalancer"
+)
+
+func (a *azureActions) ResourceDeleteAndWait(ctx context.Context, resourceID string) error {
+	idParts, err := azure.ParseResourceID(resourceID)
+	if err != nil {
+		return err
+	}
+
+	apiVersion := azureclient.APIVersion(strings.ToLower(idParts.Provider + "/" + idParts.ResourceType))
+
+	_, err = a.resources.GetByID(ctx, resourceID, apiVersion)
+	if err != nil {
+		return err
+	}
+
+	// FrontendIPConfiguration cannot be deleted with DeleteByIDAndWait (DELETE method is invalid on frontendIPConfiguration resourceID)
+	if strings.Contains(strings.ToLower(resourceID), "frontendipconfigurations") {
+		return a.deleteFrontendIPConfiguration(ctx, resourceID)
+	}
+
+	return a.resources.DeleteByIDAndWait(ctx, resourceID, apiVersion)
```

Comment:
As programmed, this is following the acceptance criteria, but I wonder if those acceptance criteria need to be modified. I think we should at least blacklist a few resources:

1. Private Endpoints - deleting this means SRE can never access the cluster again
2. Storage Accounts - potential for customer data
3. Disks - potential for customer data

Reply:
+1 to not allow PE deletion, good catch.

Reply:
I added a deny list that will prevent deletion of the Private Link Service and anything under `Microsoft.Storage`. 

I also added a e2e that attempts to delete the PLS, or is that to risky?  LMK

Reply:
Is there a `GET` that occurs on the resource before this `DELETE`? If not, you could have the E2E test use the resource ID of a non-existent PLS to test the safeguarding logic in a less risky way.

Reply:
That will work, I'll update with a bad name.